qiskit/_quantumprogram.py

```diff
@@ -387,14 +387,20 @@ def destroy_circuit(self, name):
             raise QISKitError("Can't destroy this circuit: Not present")
         del self.__quantum_program[name]
 
-    def add_circuit(self, name, quantum_circuit):
+    def add_circuit(self, name=None, quantum_circuit=None):
```

Comment:
As discussed in private, the side effect of making an argument of the public API optional when it is not really optional is not ideal ... but still, I'm not sure about the best way of avoiding it. Tricky one!